jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/jpa/PersistenceProvider.java

```diff
@@ -316,29 +320,59 @@ public JPAInitializer getInitializer(String emName, Map m){
         return JavaSECMPInitializer.getJavaSECMPInitializer(classLoader);
     }
 
+    // Package visibility is required for jUnit
     /**
-     * Need to check that the provider property is null or set for EclipseLink
+     * The Persistence bootstrap class must locate all the persistence providers using the PersistenceProviderResolver
+     * mechanism described in Section 9.3 and call createEntityManagerFactory on them in turn until an appropriate backing
+     * provider returns an EntityManagerFactory instance. A provider may deem itself as appropriate for the persistence unit
+     * if any of the following are true:
+     * <p> * Its implementation class has been specified in the provider element for that persistence unit in the persistence.xml
+     * file and has not been overridden by a different jakarta.persistence.provider property value included in the Map passed
+     * to the createEntityManagerFactory method.
+     * <p> * The jakarta.persistence.provider property was included in the Map passed to createEntityManagerFactory and the value
+     * of the property is the provider’s implementation class.
+     * <p> * No provider was specified for the persistence unit in either the persistence.xml or the property map.
+     *
+     * @since Jakarta Persistence 3.2
      */
-    public boolean checkForProviderProperty(Map properties){
```

Comment:
Hmm `public` due a JUnit test only. But it means too, that these methods will be as part of public EclipseLink API as `PersistenceProvider` is not in the internal package.

Reply:
No, there is no connection with jUnit, where pkg only is enough. See `isProviderEclipseLink` which is used in jUnit as well.
As you can see, this method was public before my change and that's the only reason why it's public now. I don't want to change public API of this class because you never know if anyone already used checkForProviderProperty int their own code.
I did change it to static, because:
1. it's not related to any content of the class instance
2. calling static metohod trough instance, e.g. `this.` works as well
3. it may be used elsewhere if needed without copy-paste